elapid/evaluate.py

```diff
@@ -0,0 +1,189 @@
+        nbins (int | list, optional): Number of classes or a list of class thresholds. Defaults to 0.
+        bin_size (float | str, optional): Width of the the bin. Defaults to 'default' which sets width as 1/10th of the fit range.
+        to_plot (bool, optional): Whether to plot the predicted-to-expected (P/E) curve. Defaults to False.
+
+    Returns:
+        dict: A dictionary with the following keys:
+            - 'F.ratio' (numpy.ndarray): The P/E ratio for each bin.
+            - 'Spearman.cor' (float): The Spearman's rank correlation coefficient between interval midpoints and F ratios.
+            - 'HS' (numpy.ndarray): The habitat suitability intervals.
+    """
+    if not isinstance(ypred, (np.ndarray, pd.Series, gpd.GeoSeries)):
+        raise TypeError("The 'ypred' parameter must be a NumPy array, Pandas Series, or GeoPandas GeoSeries.")
+    if not isinstance(yobs, (np.ndarray, pd.Series, gpd.GeoSeries)):
+        raise TypeError("The 'yobs' parameter must be a NumPy array, Pandas Series, or GeoPandas GeoSeries.")
+    if not isinstance(nbins, (int, list, np.ndarray)):
+        raise TypeError("The 'nbins' parameter must be a int, list, or 1-d NumPy array.")
+    if not isinstance(bin_size, (float, str)):
+        raise TypeError("The 'bin_size' parameter must be a float, or str ('default').")
+
+    if isinstance(bin_size, float) and (isinstance(nbins, (list, np.ndarray)) or nbins > 0):
+        raise ValueError(
+            f"Ambiguous value provided. Provide either nbins or bin_size. Cannot provide both. Provided values for nbins, bin_size are: ({nbins, bin_size})"
+        )
+
+    # Check for NaN values and issue warnings
+    if np.isnan(ypred).any():
+        warnings.warn("'ypred' contains NaN values, which will be ignored.", UserWarning)
+        ypred = ypred[~np.isnan(ypred)]
+    if np.isnan(yobs).any():
+        warnings.warn("'yobs' contains NaN values, which will be ignored.", UserWarning)
+        yobs = yobs[~np.isnan(yobs)]
+
+    ypred = np.asarray(ypred)
+    yobs = np.asarray(yobs)
+
+    if ypred.ndim != 1 or yobs.ndim != 1:
+        raise ValueError("Both 'ypred' and 'yobs' must be one-dimensional arrays.")
+
+    if len(ypred) == 0 or len(yobs) == 0:
+        raise ValueError("'ypred' or 'yobs' arrays cannot be empty.")
+
+    mini, maxi = np.min(ypred), np.max(ypred)
+
+    intervals = get_intervals(nbins, bin_size, range=[mini, maxi])
+    f_scores = np.array([boyce_index(yobs, ypred, interval) for interval in intervals])
+
+    valid = ~np.isnan(f_scores)
+    f_valid = f_scores[valid]
+
+    intervals_mid = np.mean(intervals[valid], axis=1)
+    if np.sum(valid) <= 2:
+        corr = np.nan
+    else:
+        corr, _ = spearmanr(f_valid, intervals_mid)
+
+    if to_plot:
+        plot_PE_curve(x=intervals_mid, y=f_valid)
+
+    results = {
+        "F.ratio": f_scores,
+        "Spearman.cor": corr,
+        "HS": intervals,
+    }
+
+    return results
```

Comment:
my personal preference is to avoid returning a dictionary, but instead return the three values as a tuple, so users can specify:

```python
f_ratio, cor, hs = ela.evaluate.continuous_boyce_index(presence, background)
```